starboard/shared/posix/impl/file_flush.h

```diff
@@ -36,7 +36,7 @@ bool FileFlush(SbFile file) {
     return false;
   }
 
-  return !HANDLE_EINTR(fsync(file->descriptor));
+  return !HANDLE_EINTR(fdatasync(file->descriptor));
```

Comment:
calling sync doesn't guarantee write through. Checksumming or other data integrity checks would be more robust.

Also, this fsync->fdatasync change would be really small in real world terms - it's only metadata here that may be flushed later, the data itself will still be treated the same. Best i know, we do not rely on file write timestamps and such anywhere.

Reply:
If datasync instead of sync works for chrome, we should make it work for us the same way.

Note that relying on mtime or atime can be fragile because the timestamps have different precisions depending on the filesystem. Some filesystems have only 1s or even 2s precision. Additionally, some filesystems like FAT store localtime without also storing the timezone offset, so any changes in the system timezone or daylight savings will cause the "time difference" with another timestamp to also shift. For example, once per year in the fall there is a hour during which the timestamp of an already written file can be in the future, because of the clock change.

Reply:
I wouldn't worry about the timestamp, but I would worry about file name and size. However based on the Linux man page it looks like it should be fine:

fdatasync() is similar to fsync(), but does not flush modified metadata unless that metadata is needed in order to allow a subsequent data retrieval to be correctly handled. For example, changes to st_atime or st_mtime (respectively, time of last access and time of last modification; see [stat](https://linux.die.net/man/2/stat)(2)) do not require flushing because they are not necessary for a subsequent data read to be handled correctly. On the other hand, a change to the file size (st_size, as made by say [ftruncate](https://linux.die.net/man/2/ftruncate)(2)), would require a metadata flush.

Reply:
Thanks for the +2 then, i'll merge and we'll see how it goes